master/internal/api_master.go

```diff
@@ -113,24 +113,33 @@ func (a *apiServer) PatchMasterConfig(
 			if !isValidLogLevel(req.Config.Log.Level) {
 				panic(fmt.Sprintf("invalid log level: %s", req.Config.Log.Level))
 			}
-			a.m.config.Log.Level = req.Config.Log.Level
-			a.m.config.Log.Color = req.Config.Log.Color
-			logger.SetLogrus(a.m.config.Log)
+			oldConfig := a.m.config.Load()
+			newConfig := *oldConfig
+			newConfig.Log.Level = req.Config.Log.Level
+			newConfig.Log.Color = req.Config.Log.Color
+			a.m.config.CompareAndSwap(oldConfig, &newConfig)
```

Comment:
if `CompareAndSwap` returns false we should return a 409 or retry the entire request until it succeeds. I'm fine with the former for ease and the chance we'll never hit it, really. but the latter would be nice if this was a more used endpoint.

Reply:
Done